### DISCUSSION TITLE
র‍্যাম থেকে nginx এর কনফিগ ডাম্প করা

### DISCUSSION THREAD

**Discussion by giscus** [ahmubashshir/ahmubashshir.github.io#79] [Comments]
# র‍্যাম থেকে nginx এর কনফিগ ডাম্প করা

ঘটনা বেশি দিন আগের না… গত সপ্তায় nginx এর কনফিগ এডিট করতে গিয়া sed -nEi '<regex>' /etc/nginx/sites-enabled/* মাইরা বসছিলাম… আর যা হওয়ার তাই হইল… পুরা কনফিগ উধাও… এখন ext4magic দিয়া যে রিকভার করা যাবে তাও কনফার্ম না 🙃️
কী করা যায়? দ্বারস্থ হইলাম গুরু স্তুপাধিক্যের ভাই সেবকভঙ্গের1। উনার আরেক মুরিদের কাছ থেকে জানা গেল যে nginx এর মেমোরি ডাম্প করলে ওইখানে বাইনারি ডাটার সাথে কনফিগও খিচুড়ি পাকায় থাকে… করলাম গদব(gdb) দিয়া র‍্যাম ডাম্প… strings mem_* দিয়া দেখি কনফিগ ফাইলের টেক্সট আছে… সাথে ফাংশন নেম(elf symbols) ও ডাম্প করতেছে… এইবার সাহায্যে আসলেন আরেক মুরিদ… উনি আবার আওকে(awk) ওস্তাদ… উনি যেই স্ক্রিপ্টটা দিলেন ওইটা রান করার পরে মোটামুটি সুন্দর একটা কনফিগ ডাম্প পাইলাম… বাট সব কনফিগ এক ফাইলে… পরে সব পার্ট আলাদা ফাইলে সেভ করার পরে কাজ হইল…

https://ahmubashshir.github.io/post/2023/08/26/dump-nginx-config-from-ram/

**Comment by shahriarAS** [comment]
ভাই আপনার সেন্স অফ হিউমারের প্রশংসা না করে পারছিনা। **স্তুপাধিক্য**, **সেবকভঙ্গ** 😆 😆
